RecoParticleFlow/PFRecHitProducer/interface/alpaka/CalorimeterDefinitions.h

```diff
@@ -0,0 +1,203 @@
+#ifndef RecoParticleFlow_PFRecHitProducer_interface_alpaka_CalorimeterDefinitions_h
+#define RecoParticleFlow_PFRecHitProducer_interface_alpaka_CalorimeterDefinitions_h
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+#include "DataFormats/ParticleFlowReco/interface/CaloRecHitHostCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/alpaka/CaloRecHitDeviceCollection.h"
+#include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitParamsRecord.h"
+#include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyHostCollection.h"
+#include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyRecord.h"
+#include "RecoParticleFlow/PFRecHitProducer/interface/alpaka/PFRecHitParamsDeviceCollection.h"
+#include "RecoParticleFlow/PFRecHitProducer/interface/alpaka/PFRecHitTopologyDeviceCollection.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::ParticleFlowRecHitProducer {
+
+  // Get subdetector encoded in detId to narrow the range of reference table values to search
+  // https://cmssdt.cern.ch/lxr/source/DataFormats/DetId/interface/DetId.h#0048
+  constexpr inline uint32_t getSubdet(uint32_t detId) { return ((detId >> DetId::kSubdetOffset) & DetId::kSubdetMask); }
+
+  // structs to be used as template arguments
+  struct HCAL {
+    using CaloRecHitType = HBHERecHit;
+    using CaloRecHitSoATypeHost = reco::CaloRecHitHostCollection;
+    using CaloRecHitSoATypeDevice = reco::CaloRecHitDeviceCollection;
+    using ParameterType = reco::PFRecHitHCALParamsDeviceCollection;
+    using ParameterRecordType = PFRecHitHCALParamsRecord;
+    using TopologyTypeHost = reco::PFRecHitHCALTopologyHostCollection;
+    using TopologyTypeDevice = reco::PFRecHitHCALTopologyDeviceCollection;
+    using TopologyRecordType = PFRecHitHCALTopologyRecord;
+
+    static constexpr DetId::Detector kDetectorId = DetId::Detector::Hcal;
+    static constexpr int kSubdetectorBarrelId = HcalSubdetector::HcalBarrel;
+    static constexpr int kSubdetectorEndcapId = HcalSubdetector::HcalEndcap;
+
+    static constexpr uint32_t kMaxDepthHB = 4;
+    static constexpr uint32_t kMaxDepthHE = 7;
+    static constexpr uint32_t kFirstHBRing = 1;
+    static constexpr uint32_t kLastHBRing = 16;
+    static constexpr uint32_t kFirstHERing = 16;
+    static constexpr uint32_t kLastHERing = 29;
+    static constexpr uint32_t kMaxIPhi = 72;
```

Comment:
Can these constants be taken from some existing files ?

Reply:
I cannot find a convenient place right now. The best one I see is https://github.com/fllor/cmssw/blob/ef3151a362db68a97ef10c4ff993af637cd163ef/Geometry/CaloTopology/src/HcalTopology.cc#L172-L175 but we cannot easily read from there.
@hatakeyamak, any ideas?

Reply:
Mhm, then I would suggest to move your definitions to a new file under `Geometry/CaloTopology/interface/`, and make both the alpaka PF code and the existing `HcalTopology.cc` code read them from there.

Reply:
I agree. These numbers are also duplicated elsewhere. But should this be part of this PR?

Reply:
For me it would be fine to do it in a separate PR.
In that case, please open an issue on GitHub and list there the changes that you propose to do in the separate PR.

Reply:
I don't have a good immediate suggestion. For maxDepth, it is probably better to read from:
https://github.com/cms-sw/cmssw/blob/master/Geometry/CaloTopology/src/HcalTopology.cc#L987
and actually this is in a sense not constant, because they changed after Phase 1 upgrade.
The current implementation is good for Run 3, but when we add it in official sequences in upcoming PRs, we will have to add some era dependence.
I also think it's fine to revisit in future PR (and have a github issue to keep a log.)